scipy/fft/_basic.py

```diff
@@ -161,10 +147,26 @@ def fft(x, n=None, axis=-1, norm=None, overwrite_x=False, workers=None, *,
     >>> plt.show()
 
     """
-    return (Dispatchable(x, np.ndarray),)
+    xp = array_namespace(x)
+    if is_numpy(xp):
+        return _basic_uarray.fft(x, n=n, axis=axis, norm=norm,
+                                 overwrite_x=overwrite_x, workers=workers,
+                                 plan=plan)
+    if overwrite_x:
+        raise ValueError(arg_err_msg("overwrite_x"))
+    if workers is not None:
+        raise ValueError(arg_err_msg("workers"))
+    if plan is not None:
+        raise ValueError(arg_err_msg("plan"))
+    if hasattr(xp, 'fft'):
+        if norm is None:
+            norm = 'backward'
+        return xp.fft.fft(x, n=n, axis=axis, norm=norm)
+    x = np.asarray(x)
+    y = _basic_uarray.fft(x, n=n, axis=axis, norm=norm)
+    return xp.asarray(y)
```

Comment:
@lucascolley when reviewing this again today, it stood out that there's still a lot of repeated code here. Basically this is many times the same, with only the callable names changing. In this case:
```
_basic_uarray.fft
xp.fft.fft
```
Could you try refactoring this code so it has a helper function that takes for example `xp_func` and `uarray_func` as inputs and then each public function can call that? It should shrink the diff in this file by 3x or so.